platform/default/default_file_source.cpp

```diff
@@ -151,8 +153,23 @@ class DefaultFileSource::Impl {
 
             // Get from the online file source
             if (resource.hasLoadingMethod(Resource::LoadingMethod::Network)) {
+#if MBGL_TILE_TIMING
+                using namespace std::chrono;
+                milliseconds requestBeginAt = duration_cast< milliseconds >(system_clock::now().time_since_epoch());
+#endif
                 tasks[req] = onlineFileSource.request(resource, [=] (Response onlineResponse) mutable {
                     this->offlineDatabase->put(resource, onlineResponse);
+#if MBGL_TILE_TIMING
+                    milliseconds requestEndAt = duration_cast< milliseconds >(system_clock::now().time_since_epoch());
+                    if (resource.kind == Resource::Kind::Tile) {
+                        Log::Debug(Event::HttpRequest,
```

Comment:
In general, I don't think the debug output should happen at the point of capturing "end time". Logging can be an expensive operation, busting the i-cache which can be bad in a hot loop.

Reply:
Yeah that's a fair point in general, although these two cases are not that kind of hot loop (they'll run up to maybe a few dozen times a second).

I think if you're trying to capture timing on something that's deep inside a hot loop, you'd need a different strategy which isn't logging individual events but instead building up some sort of timing data structure that can be periodically inspected (e.g. once a second you could query to ask how much time was spent in a particular area, then log that or whatever's appropriate).